packages/internal/src/generate/typeDefinitions.ts

```diff
@@ -54,6 +54,7 @@ export const generateTypeDefs = async () => {
       ...generateTypeDefGlobalContext(),
       ...generateTypeDefScenarios(),
       ...generateTypeDefTestMocks(),
+      ...generateStubStorybookTypes(),
```

Comment:
Question: do we need to do these checks against the package.json everytime? 

Perhaps we should just do it on start (rather than every time).

Also wondering: TS will generally opt for more specific types when its been defined. What happens if you just leave this in the project (via includes)?

Reply:
> Also wondering: TS will generally opt for more specific types when its been defined. What happens if you just leave this in the project (via includes)? Would the actual types from storybook override once installed?

We tried (today), but TS just showed the stub types, even after installing SB

Reply:
Ah fair, worth a shot. Upto you guys - but do you think worth moving out of the "watch" cycle and only generating this once on start of the generator?

Alternatively, we could read the package.jsons once on start (that's the slow bit)

Reply:
So I implemented caching for the file system access. And then when I went to verify that it worked as expected I realized `generateStubStorybookTypes` isn't triggered on file changes. It's only run once when the watcher is initialized. Not on any further changes.

So I actually think the code as it sits is fine.

```
watcher
  .on('ready', async () => {
    const start = Date.now()
    cliLogger('Generating full TypeScript definitions and GraphQL schemas')
    const { files, errors } = await generate()
```

It's only "on ready" we call the full `generate()` method.
On file changes we're much more selective and only call specific `generateX()` functions. And `generateStubStorybookTypes` isn't even exported, so it's not called from anywhere outside of the full `generate()` method

Reply:
Awesome!